File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    count = Counter(tangerine)
    sorted_count = count.most_common()
    
    answer = 0
    sum = 0
    
    for size, cnt in sorted_count:
        sum += cnt
        answer += 1
        if sum >= k:
            break
            
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
        public int solution(int k, int[] tangerine) {
            Map<Integer, Integer> counter = new HashMap<>();

            for (int size : tangerine) {
                counter.put(size, counter.getOrDefault(size, 0) + 1);
            }
            List<Integer> counts = new ArrayList<>(counter.values());
            counts.sort(Collections.reverseOrder());

            int sum = 0;
            int type = 0;

            for (int count : counts) {
                sum += count;
                type++;
                if (sum >= k) {
                    break;
                }
            }
            return type;
        }
    }
